# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../log4cpp/gcc43.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Description: Fix to build with gcc >= 4.3

http://log4cpp.cvs.sourceforge.net/viewvc/log4cpp/log4cpp/src/BasicLayout.cpp?r1=1.15&r2=1.16
http://log4cpp.cvs.sourceforge.net/viewvc/log4cpp/log4cpp/src/PatternLayout.cpp?r1=1.27&r2=1.28

diff -ruN log4cpp-1.0-orig/src/BasicLayout.cpp log4cpp-1.0/src/BasicLayout.cpp
--- log4cpp-1.0-orig/src/BasicLayout.cpp	2006-09-30 08:03:20.000000000 +0200
+++ log4cpp-1.0/src/BasicLayout.cpp	2010-08-16 12:52:23.544259365 +0200
@@ -11,6 +11,8 @@
 #include <log4cpp/BasicLayout.hh>
 #include <log4cpp/Priority.hh>
 #include <log4cpp/FactoryParams.hh>
+#include <memory>
+
 #ifdef LOG4CPP_HAVE_SSTREAM
 #include <sstream>
 #endif
diff -ruN log4cpp-1.0-orig/src/PatternLayout.cpp log4cpp-1.0/src/PatternLayout.cpp
--- log4cpp-1.0-orig/src/PatternLayout.cpp	2007-08-28 09:54:12.000000000 +0200
+++ log4cpp-1.0/src/PatternLayout.cpp	2010-08-16 12:51:33.856086575 +0200
@@ -23,7 +23,7 @@
 
 #include <iomanip>
 #include <ctime>
-#include <cmath>
+#include <cstdlib>
 #include "Localtime.hh"
 
 #ifdef LOG4CPP_HAVE_INT64_T
